Content/Library/docsTool/Program.fs

```diff
@@ -0,0 +1,334 @@
+﻿// Learn more about F# at http://fsharp.org
+
+open System
+open Microsoft.AspNetCore.Hosting
+open Microsoft.AspNetCore.Builder
+open Microsoft.Extensions.FileProviders
+open Microsoft.AspNetCore.Http
+open System.Net.WebSockets
+open Fake.Core
+open Fake.IO.FileSystemOperators
+open Fake.IO.Globbing.Operators
+open Fake.IO
+open Fable.React
+open Fable.React.Helpers
+open FSharp.Literate
+open System.IO
+open FSharp.MetadataFormat
+
+let refereshWebpageEvent = new Event<string>()
+
+let docsDir = FileInfo(__SOURCE_DIRECTORY__ @@ ".." @@ "docs").FullName
+let docsApiDir = docsDir @@ "api"
+let docsSrcDir = FileInfo(__SOURCE_DIRECTORY__ @@ ".." @@ "docsSrc").FullName
+
+module GenerateDocs =
+    let docsFileGlob =
+        !! (docsSrcDir @@ "**/*.fsx")
+        ++ (docsSrcDir @@ "**/*.md")
+        -- (docsSrcDir @@ "templates/*") // Don't want to generate from html templates
+
+    let render html =
+        fragment [] [
+            RawText "<!doctype html>"
+            RawText "\n"
+            html ]
+        |> Fable.ReactServer.renderToString
+    let renderWithMasterTemplate navBar titletext bodytext =
+        Master.masterTemplate "gitRepoName" navBar titletext bodytext
+        |> render
+
+    let renderWithMasterAndWrite (outPath : FileInfo) navBar titletext bodytext   =
+        let contents = renderWithMasterTemplate navBar titletext bodytext
+        IO.Directory.CreateDirectory(outPath.DirectoryName) |> ignore
+
+        IO.File.WriteAllText(outPath.FullName, contents)
+        Fake.Core.Trace.tracefn "Rendered to %s" outPath.FullName
+
+    let locateDLL name rid =
+        let lockFile = Paket.LockFile.LoadFrom (__SOURCE_DIRECTORY__ @@ ".." @@ Paket.Constants.LockFileName)
+        let packageName = Paket.Domain.PackageName name
+        let (_,package,version) =
+            lockFile.InstalledPackages
+            |> Seq.filter(fun (_,p,_) ->
+                p =  packageName
+            )
+            |> Seq.maxBy(fun (_,_,semver) -> semver)
+        Paket.NuGetCache.GetTargetUserFolder package version </> "lib" </> rid
+
+
+    let copyAssets () =
+        Shell.copyDir (docsDir </> "content")   ( docsSrcDir </> "content") (fun _ -> true)
+        Shell.copyDir (docsDir </> "files")   ( docsSrcDir </> "files") (fun _ -> true)
+
+    let generateDocs (docSourcePaths : IGlobbingPattern) githubRepoName =
+        // This finds the current fsharp.core version of your solution to use for fsharp.literate
+        let fsharpCoreDir = locateDLL "FSharp.Core" "netstandard2.0"
+
+        let parse (fileName : string) source =
+            let doc =
+                let fsharpCoreDir = sprintf "-I:%s" fsharpCoreDir
+                let runtimeDeps = "-r:System.Runtime -r:System.Net.WebClient"
```

Comment:
For the set of dependencies to typecheck/run against, we may want to steal the logic from https://github.com/fsharp/FsAutoComplete/blob/master/src/FsAutoComplete.Core/FSIRefs.fs.  This improves on the assembly-location logic from FCS (which fails per https://github.com/dotnet/fsharp/issues/7701)